src/App.tsx

```diff
@@ -1,31 +1,80 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const total = 42;
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [perPage, setPerPage] = useState<number>(5);
+  const perPageSelector = [3, 5, 10, 20];
+  const arrItems = Array.from({ length: total }, (_, i) => i + 1);
+  const items = getNumbers(1, total)
+    .map(n => `Item ${n}`);
+
+  const getPageItems = (
+    array: string[] | number[],
+    perNum: number,
+    currentNum: number,
+  ): string[] | number[] => {
+    const startIndex = (currentNum - 1) * perNum;
+    const endIndex = Math.min(startIndex + perNum, array.length);
+
+    const pageItems = array.slice(startIndex, endIndex);
+
+    return pageItems;
+  };
+
+  const selectedItemsFromTo = getPageItems(arrItems, perPage, currentPage);
+  const itemsOnPage = getPageItems(items, perPage, currentPage);
+  const totalPagesNum: number = Math.ceil(total / perPage);
+  const firstItem = selectedItemsFromTo[0];
+  const lastItem = selectedItemsFromTo[selectedItemsFromTo.length - 1];
+  const onPageChange = (event:React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    const content = event.currentTarget.textContent;
+    const currentTargetPage = parseInt(content || '0', 10);
+
+    if (currentPage !== currentTargetPage) {
```

Comment:
please add additional check bo avoid state updating
```suggestion
    if (!isNaN(currentTargetPage) && currentPage !== currentTargetPage) {
```